src/bin/rdcore/kargs.rs

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use nix::mount;
+use std::path::Path;
+
+use libcoreinst::blockdev::*;
+use libcoreinst::errors::*;
+use libcoreinst::install::*;
+
+use crate::cmdline::*;
+
+pub fn kargs(config: &KargsConfig) -> Result<()> {
+    let boot_mount = if let Some(path) = &config.boot_mount {
+        Some(Mount::from_existing(path)?)
+    } else if let Some(devpath) = &config.boot_device {
+        let devinfo = lsblk_single(Path::new(devpath))?;
+        let fs = devinfo
+            .get("FSTYPE")
+            .chain_err(|| format!("failed to query filesystem for {}", devpath))?;
+        Some(Mount::try_mount(devpath, fs, mount::MsFlags::empty())?)
+    } else {
+        unreachable!(); // checked in cmdline
```

Comment:
FYI this can take an argument to print, in case you want to drop the comment.

Reply:
I left the comment but made the string useful in case somehow the unreachable is reached.